controllers/concerts.controller.js

```diff
@@ -1,8 +1,33 @@
-const Concert = require('../models/concerts.model')
+const Concert = require('../models/concerts.model');
+const Seat = require('../models/seats.models');
+const Workshop = require('../models/workshops.model');
 
 exports.getAll = async (req, res) => {
   try {
-    res.json(await Concert.find());
+    const concerts = await Concert.find().lean();
+    const seatsFirstDay = await Seat.find({day: 1});
+    const seatsSecondDay = await Seat.find({ day: 2 });
+    const seatsThirdDay = await Seat.find({ day: 3 });
+    const workshops = await Workshop.find();
+
+    for (let concert of concerts) {
+      if (concert.day === 1) {
+        concert.ticketsLeft = 50 - seatsFirstDay.length
+      } else if (concert.day === 2) {
+        concert.ticketsLeft = 50 - seatsSecondDay.length
+      } else if (concert.day === 3) {
+        concert.ticketsLeft = 50 - seatsThirdDay.length
+      }
+
+      concert.workshops = [];
+
+      for (let workshop of workshops) {
+        if (concert._id == workshop.concertId) {
+          concert.workshops.push(workshop.name);
```

Comment:
Może zamiast pobierać wszystkie workshopy na starcie, to wyszukaj tylko te workshopy, które pasują do id koncertu, np. 

`const workshops = await Workshop.find({concertId: concert._id});`

Reply:
Dobry pomysł. W sumie to można to też wykorzystać dla Seats, żeby nie pobierać oddzielnie dla każdego dnia.